jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -79,6 +84,15 @@ public void readValues( Object object ) {
                 throw new RuntimeException( "Error while reading field " + field, e );
             }
         }
+        for(Field field : getComposedStageFields(object ) ) {
+            try {
+                Object substage = field.get(object);
+                readValues( substage );
```

Comment:
Ok, I now think I understood your approach. It is a bit different to how I would have implemented it. You essentially treat the state of the substages as part of the parent stage.